File: Python3/1 - 200/62 Unique Paths.py
# 36ms, 100.0%
class Solution:
    def uniquePaths(self, m, n):
        fac1, fac2 = m + n - 2, min(m, n) - 1
        res1 = res2 = 1
        for i in range(1, fac2 + 1):
            res1, res2 = res1 * fac1, res2 * i
            fac1 = fac1 - 1
        return res1 // res2
        

File: 1 - 200/62 Unique Paths.cc
// 2ms, 28.59%
class Solution {
public:
    int uniquePaths(int m, int n) {
        int fac1 = m + n - 2, fac2 = min(m, n) - 1;
        double res = 1;
        for (; fac2 > 0; --fac1, --fac2)
            res *= double(fac1) / fac2;
        return int(res + 0.5);
    }
};
